nmigen/hdl/ir.py

```diff
@@ -532,6 +532,13 @@ def prepare(self, ports=None, missing_domain=lambda name: ClockDomain(name)):
         if ports is None:
             fragment._propagate_ports(ports=(), all_undef_as_ports=True)
         else:
+            # typecheck ports is either list or tuple (#362)
+            if not isinstance(ports, tuple) and not isinstance(ports, list):
+                msg = "`ports` must be either a list or a tuple, not {!r}"\
+                        .format(type(ports).__name__)
```

Comment:
The nMigen convention is to use repr of the value in the error, not the name of the type. The reason is that it is often easier to recognize where `1024` came from than an abstract `int`. I would accept a PR that changes every diagnostic to report *both*, but in absence of that let's consistently use repr of the value.